### ISSUE TITLE
(aws-rds): create an aurora read replica cluster from an rds instance

### ISSUE THREAD

**Issue by jtmthf** [aws/aws-cdk#17080]
### Description

The ability to create an aurora DatabaseCluster that is a read replica of an existing RDS instance.

### Use Case

As described in the aurora migration docs https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/AuroraPostgreSQL.Migrating.html#AuroraPostgreSQL.Migrating.RDSPostgreSQL.Replica, one proposed approach is to create an aurora read replica cluster of an existing rds instance. This is currently not possible using L2 constructs.

### Proposed Solution

This could be done by either adding a `sourceDatabaseInstance` prop to `DatabaseCluster` in the form of `sourceDatabaseInstance?: IDatabaseInstance` or by creating a `DatabaseClusterReadReplica` construct which creates an aurora cluster from an existing rds instance.

### Other information

#6052 is similar but seems to imply creating a read replica instance from an existing cluster.

Additionally I did try to update the underlying `CfnDatabaseCluster` but it is not exposed as a property of `DatabaseCluster`

### Acknowledge

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

**Comment by lielran** [comment]
It seems that is not possible to link `DatabaseCluster` to `DatabaseInstanceReadReplica` not because of the API.
while `DatabaseCluster` provides `instances` number property, you couldn't change the instance type and size to be difer from the "main"/writer

I've tried to use this:
```typescript
 const cluster = new rds.DatabaseCluster(this, 'Cluster', {
          // ...
        });
      const db = DatabaseInstance.fromDatabaseInstanceAttributes(this, 'DB', {
                instanceIdentifier: cluster.instanceIdentifiers[0],
                instanceEndpointAddress: 'garbage value', // Can be an arbitrary value
                port: props.dbProps.databasePort,
                securityGroups: [securityGroup],
            });
            new DatabaseInstanceReadReplica(this, 'readonly-for-application', {
                instanceIdentifier: `${props.serviceName}-${props.stage}-application-readonly`,
                sourceDatabaseInstance: db,
                instanceType: InstanceType.of(InstanceClass.BURSTABLE3, InstanceSize.MEDIUM),
                vpcSubnets: vpc.selectSubnets({ subnets: vpc.isolatedSubnets.concat(vpc.privateSubnets) }),
                vpc,
            });
```
but got this error:

```
dev-provisioned-database | 6:56:48 AM | CREATE_FAILED        | AWS::RDS::DBInstance                        | readonly-for-internal-usage (readonlyforinternalusage2A74F57B) Cannot create a read replica for a DB Instance that belongs to a DB Cluster (Service: AmazonRDS; Status Code: 400; Error Code: InvalidParameterCombination; Request ID: 73e364c5-ed07-4b07-825b-85a960c43e4e; Proxy: null)
```

any workaround for that?

**Comment by skinny85** [comment]
Yes, that is not supported, until the feature request detailed in this issue, the `DatabaseClusterReadReplica`, is created.

For now, you need to use the low-level `CfnDBInstance` class to get this working.

**Comment by spolired** [comment]
Hi @[skinny85](https://github.com/skinny85) Is there an example on using <code>CfnDBInstance<code> to create a Aurora Cluster ReadReplica

**Comment by skinny85** [comment]
I don't have an example handy, but I would Google on how to do it in CloudFormation, and, once I see the template, it should be fairly easy to replicate it in CDK code (or use the [CFN Include](https://docs.aws.amazon.com/cdk/api/v1/docs/cloudformation-include-readme.html) module).

**Comment by alexsavio** [comment]
I don't recommend using the CfnDBInstance because, as far as I can see, you would have trouble fetching the names and ids of the resources defined with the high-level syntax, such as ParameterGroup.

**Comment by moltar** [comment]
Might be done via #33280

**Comment by aemada-aws** [comment]
Duplicate of https://github.com/aws/aws-cdk/issues/33280 which is solved https://github.com/aws/aws-cdk/pull/33471 via the `replicationInstanceSource` prop.
